src/test/groovy/com/ofg/uptodate/finder/http/ProxySettingsResolverSpec.groovy

```diff
@@ -0,0 +1,57 @@
+package com.ofg.uptodate.finder.http
+import com.ofg.uptodate.UptodatePluginExtension
+import spock.lang.Specification
+
+class ProxySettingsResolverSpec extends Specification {
+    
+    private static final String SYS_PROP_HTTP_PROXY_HOSTNAME = 'http-proxy-sys-prop'
+    private static final int SYS_PROP_HTTP_PROXY_PORT = 15000
+    private static final String SYS_PROP_HTTPS_PROXY_HOSTNAME = 'https-proxy-sys-prop'
+    private static final int SYS_PROP_HTTPS_PROXY_PORT = 15100
+    private static final String PLUGIN_CONFIG_PROXY_HOSTNAME = 'http-proxy-plugin-config'
+    private static final int PLUGIN_CONFIG_PROXY_PORT = 15200
+    private static final String PLUGIN_CONFIG_PROXY_SCHEME = 'http'
+    
+    def "'http.*' system properties should have the highest priority when resolving proxy configuration"() {
+        given: "proxy is configured via 'http,*' system properties"
+            String prevHttpProxyHost = System.setProperty('http.proxyHost', SYS_PROP_HTTP_PROXY_HOSTNAME)
+            String prevHttpProxyPort = System.setProperty('http.proxyPort', SYS_PROP_HTTP_PROXY_PORT.toString())
+        and: "proxy is configured via 'https,*' system properties"
+            String prevHttpsProxyHost = System.setProperty('https.proxyHost', SYS_PROP_HTTPS_PROXY_HOSTNAME)
+            String prevHttpsProxyPort = System.setProperty('https.proxyPort', SYS_PROP_HTTPS_PROXY_PORT.toString())
+        and: "proxy is configured via plugin configuration"
+            UptodatePluginExtension uptodatePluginExtension = new UptodatePluginExtension(proxyHostname: PLUGIN_CONFIG_PROXY_HOSTNAME, proxyPort: PLUGIN_CONFIG_PROXY_PORT, proxyScheme: PLUGIN_CONFIG_PROXY_SCHEME)
+        expect:
+            new ProxySettings(hostname: SYS_PROP_HTTP_PROXY_HOSTNAME, port: SYS_PROP_HTTP_PROXY_PORT, scheme: 'http') == new ProxySettingsResolver(uptodatePluginExtension).resolve()
+        cleanup:
+            System.setProperty('http.proxyHost', prevHttpProxyHost ?: '')
```

Comment:
This extension cannot be used at the moment because of the version of spock we use in the project, 0.7-groovy-1.8. I guess updating it to 1.0 would also require upgrading the gradle version, because as it is right now, I get:
```
The Spock compiler plugin cannot execute because Spock 1.0.0-groovy-2.4 is not compa
tible with Groovy 1.8.6. For more information, see http://versioninfo.spockframework.org

Spock artifact: [...]/.gradle/caches/modules-2/files-2.1/org.spockframework/spock-core/1.0-groovy-2.4/ceaa8b69f274ed3de24da3e6a6c86f673b426d1a/spock-core-1.0-groovy-2.4.jar
Groovy artifact: [...]/.gradle/wrapper/dists/gradle-1.12-bin/64p5p2nte80b6rt6bb068pabi6/gradle-1.12/lib/groovy-all-1.8.6.jar
```